### DISCUSSION TITLE
How to do: `zstd --adapt`

### DISCUSSION THREAD

**Discussion by zwimer** [indygreg/python-zstandard#239] [Q&A] [answered]
Does this support `zstd`'s `--adapt` option anywhere?

Specifically (from the man page)
> --adapt[=min=#,max=#] : zstd will dynamically adapt compression level to perceived I/O conditions. Compression level adaptation can be observed live by using command -v. Adaptation can be constrained between supplied min and max levels. The feature works when combined with multi-threading and --long mode. It does not work with --single-thread. It sets window size to 8 MB by default (can be changed manually, see wlog). Due to the chaotic nature of dynamic adaptation, compressed result is not reproducible. note : at the time of this writing, --adapt can remain stuck at low speed when combined with multiple worker threads (>=2).

**Comment by indygreg** [comment, marked as answer]
No.

The `--adapt` functionality is implemented outside of libzstd and therefore is not available for us to use unless we reimplement the core logic. That would be a big lift and not something I have cycles to support.

If the adapt functionality were available to libzstd, I'd love to support it.

I'm sure this topic has come up in the upstream project. But I'm not aware of any specific conversations and opinions from the zstd maintainers.
